magma/interface.py

```diff
@@ -491,6 +494,12 @@ def __getattr__(self, key):
             return self._ports[key]
         return super().__getattribute__(key)
 
+    def flip(self):
```

Comment:
let's add a utility function called fields which is basically like

```python
def fields(self) -> Dict[str, Kind]:
    return {name: T for name, T in zip(self._decl[::2], self._decl[1::2])}
```

this will probably be useful to be public for users as well